sui_core/src/authority/authority_store.rs

```diff
@@ -761,7 +775,7 @@ impl<
                     })
                     .collect();
                 self.lock_service
-                    .initialize_locks(new_locks_to_init)
+                    .initialize_locks(&new_locks_to_init, false /* force_reset */)
```

Comment:
Um, this one should be marked true, it is the normal case in a transaction and we need to check for not previous initialization

Reply:
Are you sure? Say if we are executing an older cert, which would lead to re-initializing the locks of the new objects, which may override a pending transaction that's trying to mutate them?

Reply:
@velvia Please let me know what you think. I think we shouldn't change the semantics here. The only case where we can safely use this flag is from the gateway, not in the validator.

Reply:
Oh, never mind.  I misunderstood your comment to mean that false means to force a reset.   You are right that it should be false, which should be a default.   I think it's better to remove the comment as its confusing.